ingestion_server/ingestion_server/indexer.py

```diff
@@ -309,135 +297,89 @@ def replicate(self, table, dest_index, query):
+        :param index_suffix: the unique suffix to add to the index name
         """
-        while True:
-            log.info("Listening for updates...")
-            try:
-                for table in self.tables_to_watch:
-                    self._index_table(table)
-            except ESConnectionError:
-                self.es = elasticsearch_connect()
-            time.sleep(poll_interval)
 
-    def reindex(self, model_name: str, distributed=None):
+        if not index_suffix:
+            index_suffix = uuid.uuid4().hex
+        if not table_name:
+            table_name = model_name
+        destination_index = f"{model_name}-{index_suffix}"
+
+        log.info(
+            f"Creating index {destination_index} for model {model_name} "
+            f"from table {table_name}."
+        )
+        self.es.indices.create(
+            index=destination_index,
+            body=index_settings(model_name),
+        )
+
+        log.info("Running distributed index using indexer workers.")
+        self.active_workers.value = int(True)
+        schedule_distributed_index(
+            database_connect(), model_name, table_name, destination_index, self.task_id
+        )
+
+    def update_index(self, model_name: str, index_suffix: str, since_date: str, **_):
         """
-        Copy contents of the database to a new Elasticsearch index. Create an
-        index alias to make the new index the "live" index when finished.
+        Update index based on changes in the database after the given date.
+
+        :param model_name: the name of the media type
+        :param index_suffix: the unique suffix of the index to update
+        :param since_date: the date after which to update the records
         """
-        suffix = uuid.uuid4().hex
-        destination_index = f"{model_name}-{suffix}"
-        if distributed is None:
-            distributed = config("ENVIRONMENT", default="local") != "local"
-        if distributed:
-            self.es.indices.create(
-                index=destination_index, body=index_settings(model_name)
-            )
-            self.active_workers.value = int(True)
-            schedule_distributed_index(
-                database_connect(), destination_index, self.task_id
-            )
-        else:
-            self._index_table(model_name, dest_idx=destination_index)
-            slack.verbose(
-                f"`{model_name}`: Elasticsearch reindex complete | "
-                f"_Next: promote index as primary_"
-            )
-            self.go_live(destination_index, model_name)
 
-    def update(self, model_name: str, since_date):
-        log.info(f"Updating index {model_name} with changes since {since_date}")
+        destination_index = f"{model_name}-{index_suffix}"
+
+        log.info(f"Updating index {destination_index} with changes since {since_date}.")
         deleted, mature = get_existence_queries(model_name)
```

Comment:
Do we need to pass `table_name` here now too?

Reply:
So existence queries will reuse the model name as the table name if you don't explicitly pass it (which it basically accurate).

Reply:
It will definitely work now, since the model names all match the table names. I'm wondering if this would break if we introduced a model/table where that was not the case?